src/php/Interop/ExportFinder/FunctionsToExportFinder.php

```diff
@@ -54,17 +55,30 @@ public function findInPaths(): array
      */
     private function loadAllNsFromPaths(): void
     {
+        $this->commandFacade->registerExceptionHandler();
+
         $namespaceFromDirectories = $this->buildFacade
             ->getNamespaceFromDirectories($this->exportDirs);
 
-        $namespaces = array_values(array_map(fn (NamespaceInformation $info) => $info->getNamespace(), $namespaceFromDirectories));
+        $namespaces = array_values(array_map(
+            static fn (NamespaceInformation $info) => $info->getNamespace(),
+            $namespaceFromDirectories
+        ));
+
+        $dependenciesForNamespace = [];
+        foreach ($namespaces as $namespace) {
+            $dependenciesForNamespace[] = $this->buildFacade->getDependenciesForNamespace(
+                [
+                    ...$this->commandFacade->getSourceDirectories(),
+                    ...$this->commandFacade->getVendorSourceDirectories(),
+                ],
+                [$namespace, 'phel\\core']
```

Comment:
I don't think, we need the foreach loop here. Instead we can just past `[...$namespaces, 'phel\\core']` as argument of the `getDependenciesForNamespace`. Otherwise `$allNamespace` will contain duplicates that can lead to unexpected behavior.

Reply:
Right! Good catch. I will update it.